src/components/Article.js

```diff
@@ -14,30 +14,34 @@ const ArticleBody = memo(({ content }) => {
   )
 })
 
+const ArticleAudioPopup = ({ closeFn }) => {
+  useSoftkey('ArticleAudioPopup', {
+    center: 'ClosePopup',
+    onKeyCenter: closeFn
+  }, [])
+  const style = {
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    bottom: '30px',
+    top: 100,
+    backgroundColor: 'pink',
+    zIndex: 999
+  }
+  return (
+    <div class='popup' style={style}>
+      <h1>This is a popup on top of the article page</h1>
+    </div>
+  )
+}
+
 const ArticleSection = ({
   lang, imageUrl, title, description, hasActions, content, showToc
 }) => {
   const i18n = useI18n()
-  const softkey = useSoftkey()
   const actionsRef = useRef()
-  const [, setNavigation, getCurrent] = useNavigation(actionsRef, 'x')
-
-  useEffect(() => {
-    softkey.dispatch({ type: 'setOnKeyCenter', event: onKeyCenter })
-  }, [])
-
-  useEffect(() => {
-    // todo: the enter key should only be available on the
-    // first page of the first section
-    softkey.dispatch({
-      type: 'setCenterText',
-      value: hasActions ? i18n.i18n('centerkey-select') : ''
-    })
-    if (hasActions) {
-      setNavigation(0)
-    }
-  }, [hasActions])
-
+  const [,, getCurrent] = useNavigation('Article', actionsRef, 'x')
+  const [isVisible, setVisible] = useState(false)
```

Comment:
should this rename to which component is visible? or if this is the temporary state, then make the comment so we know it when we reach this again.

Reply:
Done